### ISSUE TITLE
How to verify the model after trainning?

### ISSUE THREAD

**Issue by luyunxi** [starcolon/photo-auto-balance#20]
After training the data, I used "python3 loader.py --enhance --dir ./data/sample" to enhance the images. And I got bad resluts which are black images.  Is there some mistake.

**Comment by ida03001** [comment]
Me too. After training the data, I got training loss and validation loss as below: 
...Training Loss   :  0.976349,1.018974,16.172292,0.999951
...Validation Loss :  1.090888,1.002220,15.597400,0.990669.
I used "python3 loader.py --train --limit 30 --batch 50 --dir /path/to/trainset/" to train the model.
Is there some mistake? Thanks!

**Comment by ida03001** [comment]
After training the data, I used "python3 loader.py --enhance --dir ./data/sample" to enhance the images.
Print these:
> /usr/local/lib/python3.5/dist-packages/sklearn/cross_validation.py:41: DeprecationWarning: This module was deprecated in version 0.18 in favor of the model_selection module into which all the refactored classes and functions are moved. Also note that the interface of the new CV iterators are different from that of this module. This module will be removed in 0.20.
  "This module will be removed in 0.20.", DeprecationWarning)
=====================
Loader started ...
=====================
Executing function...
Loading samples...
(30, 3, 240, 240)
Loading the models at data/raw/model.cnn
...Loading part #0
...[done]
...Loading part #1
...[done]
...Loading part #2
...[done]
...Loading part #3
...[done]
[All loaded]
Predicting 30 samples...
WARNING (theano.tensor.blas): We did not found a dynamic library into the library_dir of the library we use for blas. If you use ATLAS, make sure to compile it with dynamics library.
[[  0.00000000e+00  -2.21524188e-15   1.30104261e-16   0.00000000e+00]
 [  0.00000000e+00   2.62064676e-14   1.30104261e-16   0.00000000e+00]
 [  0.00000000e+00  -2.21524188e-15   1.30104261e-16   0.00000000e+00]
 [  0.00000000e+00  -2.21524188e-15   1.30104261e-16   0.00000000e+00]
 [  0.00000000e+00   2.62064676e-14   1.30104261e-16   0.00000000e+00]
 [  0.00000000e+00  -2.21524188e-15   1.30104261e-16   0.00000000e+00]
 [  0.00000000e+00   2.62064676e-14   1.30104261e-16   0.00000000e+00]
 [  0.00000000e+00   2.62064676e-14   1.30104261e-16   0.00000000e+00]
 [  0.00000000e+00  -2.21524188e-15   1.30104261e-16   0.00000000e+00]
 [  0.00000000e+00   2.62064676e-14   1.30104261e-16   0.00000000e+00]
 [  0.00000000e+00  -2.21524188e-15   1.30104261e-16   0.00000000e+00]
 [  0.00000000e+00   2.62064676e-14   1.30104261e-16   0.00000000e+00]
 [  0.00000000e+00  -2.21524188e-15   1.30104261e-16   0.00000000e+00]
 [  0.00000000e+00   2.62064676e-14   1.30104261e-16   0.00000000e+00]
 [  0.00000000e+00   2.62064676e-14   1.30104261e-16   0.00000000e+00]
 [  0.00000000e+00  -2.21524188e-15  -9.19403442e-17   0.00000000e+00]
 [  0.00000000e+00   2.62064676e-14   1.30104261e-16   0.00000000e+00]
 [  0.00000000e+00   2.62064676e-14   1.30104261e-16   0.00000000e+00]
 [  0.00000000e+00  -2.21524188e-15   1.30104261e-16   0.00000000e+00]
 [  0.00000000e+00  -2.21524188e-15  -9.19403442e-17   0.00000000e+00]
 [  0.00000000e+00  -2.21524188e-15  -9.19403442e-17   0.00000000e+00]
 [  0.00000000e+00   2.62064676e-14   1.30104261e-16   0.00000000e+00]
 [  0.00000000e+00   2.62064676e-14   1.30104261e-16   0.00000000e+00]
 [  0.00000000e+00  -2.21524188e-15  -9.19403442e-17   0.00000000e+00]
 [  0.00000000e+00  -2.21524188e-15   1.30104261e-16   0.00000000e+00]
 [  0.00000000e+00  -2.21524188e-15  -9.19403442e-17   0.00000000e+00]
 [  0.00000000e+00  -2.21524188e-15   1.30104261e-16   0.00000000e+00]
 [  0.00000000e+00   2.62064676e-14   1.30104261e-16   0.00000000e+00]
 [  0.00000000e+00   2.62064676e-14   1.30104261e-16   0.00000000e+00]
 [  0.00000000e+00   2.62064676e-14   1.30104261e-16   0.00000000e+00]]
(30, 4)
> 
Where did i make mistake? Thanks!

**Comment by tao-pr** [comment]
the errors look really big to me. given your parameters i suspect two things:
- the limit of the # of samples is way too low. try increasing the limit from below 100 to thousands. i suggest increasing the number of the samples a lot larger will help. also, better to increase the number of the `permutation` too so it generates more training data given your training images. BUT, this will slow down the whole iteration, be patient.
- try running on larger number of iters, i would keep an eye on the model errors on each iteration and check if it actually decreases. for too small dataset, you may often get stuck and the errors won't go down at all during the iterations.

**Comment by ida03001** [comment]
@starcolon Thank you for your reply and advice. I will try. Thanks!
